azkaban-common/src/main/java/azkaban/executor/AbstractExecutorManagerAdapter.java

```diff
@@ -0,0 +1,436 @@
+package azkaban.executor;
+
+
+import azkaban.Constants;
+import azkaban.Constants.ConfigurationKeys;
+import azkaban.event.EventHandler;
+import azkaban.flow.FlowUtils;
+import azkaban.metrics.CommonMetrics;
+import azkaban.project.Project;
+import azkaban.project.ProjectWhitelist;
+import azkaban.utils.FileIOUtils.LogData;
+import azkaban.utils.Pair;
+import azkaban.utils.Props;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AbstractExecutorManagerAdapter extends EventHandler implements
```

Comment:
One alternative way of encapsulating this code which doesn't introduce inheritance could be declaring an intermediate interface with package access only which extends the `ExecutorManagerAdapter` interface and it is implemented by the `ExecutionController` and `ExecutorManager` classes. Something like:

```
interface IntermediateExecutorManagerAdapter extends ExecutorManagerAdapter {
  ExecutorLoader getExecutorLoader();
  
  default ExecutableFlow getExecutableFlow(final int execId)
      throws ExecutorManagerException {
    return getExecutorLoader().fetchExecutableFlow(execId);
  }
}
```

```
public class ExecutionController extends EventHandler implements IntermediateExecutorManagerAdapter {
@Override
  public ExecutorLoader getExecutorLoader() {
    return this.executorLoader;
  }
}
```

Reply:
Thank you. Let me think about this change and cover it in next PR. I want to keep this PR simple to reduce the amount of effect it could cause.